### PR TITLE
chore: Bump coverage Python and remove Windows coverage tests

### PR DESCRIPTION
**Pull request by jamesclark-Zapata**
# The problem

We're targeting Python 3.11 so we need to test on it.
Windows tests are broken and we don't plan on fixing them.

# This PR's solution

Bump Python version.
Remove Windows tests

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/imp[rovement]/int[ernal]/docs>[!]:`
- [x] The PR is linked to a JIRA ticket (if there's no suitable ticket, check the box).

### CODE REVIEW

File: tests/runtime/ray/test_integration.py [hunk 1]
```diff
@@ -698,6 +698,9 @@ class Test3rdPartyLibraries:
     # likely to be different from the one we're using for development. The SDK shows a
     # warning when deserializing a workflow def like this.
     @pytest.mark.filterwarnings("ignore::orquestra.sdk.exceptions.VersionMismatch")
+    @pytest.mark.skipif(
+        sys.version_info < (3, 11), reason="Pickle internals changed in Python 3.11"
```
Comment (alexjuda) [inline tests/runtime/ray/test_integration.py:702]: If there's inter-version mismatch I think we should focus on the newest one.
* Have you considered updating the test data from `3.11` and skipping the test if running on `3.10` and below?
* If you agree, but that's too much of a task for now, could you add a separate ticket for it?

**Review by BenjaminMummery — APPROVED** [review]
LGTM

**Review by alexjuda — APPROVED** [review]
Accepted with a nitpick

File: tests/runtime/ray/test_integration.py (hunk 1 above)
Comment (jamesclark-Zapata) [reply tests/runtime/ray/test_integration.py:702]: That's what this does

Comment (alexjuda) [reply tests/runtime/ray/test_integration.py:702]: Holy moly you're right

**Pull request merged**